protected/controllers/AdminSampleController.php

```diff
@@ -2,126 +2,108 @@
-			'model'=>$this->loadModel($id),
-		));
-	}
-
-	/**
-	 * Creates a new model.
-	 * If creation is successful, the browser will be redirected to the 'view' page.
-	 */
-	public function actionCreate()
-	{
-		$model=new Sample;
-
-		// Uncomment the following line if AJAX validation is needed
-		// $this->performAjaxValidation($model);
-               
-		if(isset($_POST['Sample']))
-		{
-			$model->attributes=$_POST['Sample'];
-                        $model->name = $_POST['Sample']['name'];
-                        
-                        $array = explode(":",$_POST['Sample']['species_id']);
-                        $tax_id=$array[0];
-                        $species = Species::model()->findByAttributes(array('tax_id' => $tax_id));
-                        $model->species_id=$species->id;
-                        $model->attributesList = $_POST['Sample']['attributesList'];
-                        
-            //              if(strstr($_POST['Sample']['code'],':'))
-            //     {
-            //     //$attribute_temp=null;
-            //     $temp=explode(':', $_POST['Sample']['code']);
-            //     if($temp[0]=='SAMPLE')
-            //     {
-            //         $xmlpath=  'http://www.ebi.ac.uk/ena/data/view/'."$temp[1]".'&display=xml';
-            //         $allfile= simplexml_load_file($xmlpath);
-                    
-                   
-                    
-            //     foreach ($allfile->SAMPLE->SAMPLE_ATTRIBUTES->SAMPLE_ATTRIBUTE as $child)
-            //     {
-            //         if($child->TAG=='Sample type'||$child->TAG=='Time of sample collection'||$child->TAG=='Habitat'||$child->TAG=='Sample extracted from')
-            //             $attribute_temp.= $child->TAG." = "."\"".$child->VALUE."\", ";
-            //     }
-            //     //$attribute_temp.="Description = "."\"".$allfile->SAMPLE->DESCRIPTION."\", ";
-                      
-            //     }
-            //     //$model->s_attrs=$attribute_temp;
-            // }
-                    if($model->save()) {
-                        $this->updateSampleAttributes($model);
-                        $this->redirect(array('view','id'=>$model->id));
-                    }
-		}
-		$this->render('create',array(
-			'model'=>$model,
-		));
-	}
-         
-        
-        public function actionCreate1() {
-            $model = new Sample;
+            array('deny',  // deny all users
+                'users' => array('*'),
+            ),
+        );
+    }
+
+    /**
+     * Displays a particular model.
+     * @param integer $id the ID of the model to be displayed
+     */
+    public function actionView($id)
+    {
+        $this->render('view', array(
+            'model' => $this->loadModel($id),
+        ));
+    }
+
+    /**
+     * Creates a new model.
+     * If creation is successful, the browser will be redirected to the 'view' page.
+     */
+    public function actionCreate()
+    {
+        $model = new Sample();
```

Comment:
Instead of `$model`, I would name it `$sample` so I know what type of model object I am dealing with lower down.

Reply:
This $model is declared inside  `AdminSampleController` class, which makes it clear that it is referred to the `Sample` model, and I think all `$model` in this controller class is referred to the `Sample` model as well.
So, I would better not change it, otherwise many `$model` in this class have to be re-declared.

Reply:
This $model is declared inside  `AdminSampleController` class, which makes it clear that it is referred to the `Sample` model, and I think all `$model` in this controller class is referred to the `Sample` model as well.
So, I would better not change it, otherwise many `$model` in this class have to be re-declared.